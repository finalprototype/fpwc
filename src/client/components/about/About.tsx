import React from 'react';

import Footer from '../ui/Footer';
import PageContent from '../ui/PageContent';
import FmvBackground from '../ui/FmvBackground';
import Video from '../../videos/sea.mp4';
import Fallback from '../../images/bkgds/sea.jpg';

const About: React.FunctionComponent = () => (
  <FmvBackground
    videoSource={Video}
    imageFallback={Fallback}
  >
    <PageContent flex centered>
      - in development -
    </PageContent>
    <Footer fixed />
  </FmvBackground>
);

export default About;
