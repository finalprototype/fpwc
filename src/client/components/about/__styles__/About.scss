.container {
  position: relative;
}

.content {
  position: relative;

  h1, h2, h3, h4 {
    text-transform: uppercase;
  }

  section {
    position: relative;
    overflow: hidden;
    &:not(last-child) {
      margin-bottom: 2em;
    }
  }
}

.face {
  width: 128px;
  height: 128px;
  float: right;
}

.details {
  ul {
    list-style-type: none;
  }

  li {
    margin: 12px 0;

    h3, h4, h5 {
      font-style: italic;
    }
  }
}

.game-links {
  text-align: center;
  button {
    margin: 8px;
  }
}

.nintendo {
  text-align: center;
  font-size: calc(max(min(6vw, 50px), 18px));
}

.friendcode {
  font-weight: 700;
}
