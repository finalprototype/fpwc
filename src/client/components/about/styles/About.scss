.container {
  position: relative;
}

.content {
  position: relative;

  h1, h2 {
    text-transform: uppercase;
  }

  section {
    &:not(last-child) {
      margin-bottom: 2em;
    }
  }
}
