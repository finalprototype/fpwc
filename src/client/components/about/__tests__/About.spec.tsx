import React from 'react';
import { shallow, mount } from 'enzyme';
import routeData from 'react-router';
import { BrowserRouter } from 'react-router-dom';

import About from '../About';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

const mockLocation = {
  pathname: '/about',
  hash: '',
  search: '',
  state: '',
};

beforeEach(() => {
  jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
});

afterEach(() => {
  (routeData.useLocation as jest.Mock).mockRestore();
});

describe('About', () => {
  it('renders site content by default', () => {
    const wrapper = shallow(
      <About />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders me content', () => {
    (routeData.useLocation as jest.Mock).mockReturnValueOnce({
      ...mockLocation,
      pathname: '/about/me',
    });
    const wrapper = shallow(
      <About />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders site content', () => {
    (routeData.useLocation as jest.Mock).mockReturnValueOnce({
      ...mockLocation,
      pathname: '/about/site',
    });
    const wrapper = shallow(
      <About />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches background update', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0);
    mount(
      <BrowserRouter>
        <About />
      </BrowserRouter>
    );
    expect(mockDispatch).toHaveBeenCalledTimes(1);
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: 'BACKGROUND_CHANGE',
      payload: {
        imageSource: 'mock-file',
        videoSource: 'mock-file',
      },
    });
  });
});
