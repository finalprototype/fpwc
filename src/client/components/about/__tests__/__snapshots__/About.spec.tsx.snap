// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders me content 1`] = `
<Fragment>
  <PageContent
    Sidebar={
      <SideNav>
        <NavItem
          block={true}
          isActive={false}
          label="About The Site"
          route="/about/site"
          type="right"
        />
        <NavItem
          block={true}
          isActive={true}
          label="About Me"
          route="/about/me"
          type="right"
        />
      </SideNav>
    }
    centered={false}
    className="content"
    flex={true}
    full={false}
  >
    <AboutMeContent />
  </PageContent>
  <Footer
    fixed={true}
  />
</Fragment>
`;

exports[`About renders site content 1`] = `
<Fragment>
  <PageContent
    Sidebar={
      <SideNav>
        <NavItem
          block={true}
          isActive={true}
          label="About The Site"
          route="/about/site"
          type="right"
        />
        <NavItem
          block={true}
          isActive={false}
          label="About Me"
          route="/about/me"
          type="right"
        />
      </SideNav>
    }
    centered={false}
    className="content"
    flex={true}
    full={false}
  >
    <AboutSiteContent />
  </PageContent>
  <Footer
    fixed={true}
  />
</Fragment>
`;

exports[`About renders site content by default 1`] = `
<Fragment>
  <PageContent
    Sidebar={
      <SideNav>
        <NavItem
          block={true}
          isActive={false}
          label="About The Site"
          route="/about/site"
          type="right"
        />
        <NavItem
          block={true}
          isActive={false}
          label="About Me"
          route="/about/me"
          type="right"
        />
      </SideNav>
    }
    centered={false}
    className="content"
    flex={true}
    full={false}
  >
    <AboutSiteContent />
  </PageContent>
  <Footer
    fixed={true}
  />
</Fragment>
`;
